evalml/pipelines/components/transformers/imputers/time_series_imputer.py

```diff
@@ -0,0 +1,175 @@
+"""Component that imputes missing data according to a specified timeseries-specific imputation strategy."""
+import pandas as pd
+
+from evalml.pipelines.components.transformers import Transformer
+from evalml.utils import infer_feature_types
+
+
+class TimeSeriesImputer(Transformer):
+    """Imputes missing data according to a specified timeseries-specific imputation strategy.
+
+    This Transformer should be used after the `TimeSeriesRegularizer` in order to impute the missing values that were
+    added to X and y (if passed).
+
+    Args:
+        categorical_impute_strategy (string): Impute strategy to use for string, object, boolean, categorical dtypes.
+                                              Valid values include "backwards_fill" and "forwards_fill". Defaults to "forwards_fill".
+        numeric_impute_strategy (string): Impute strategy to use for numeric columns. Valid values include
+                                          "backwards_fill", "forwards_fill", and "interpolate". Defaults to "interpolate".
+        target_impute_strategy (string): Impute strategy to use for the target column. Valid values include "backwards_fill",
+                                         "forwards_fill", and "interpolate". Defaults to "forwards_fill".
+        random_seed (int): Seed for the random number generator. Defaults to 0.
+    """
+
+    modifies_features = True
+    modifies_target = True
+    training_only = True
+
+    name = "Time Series Imputer"
+    hyperparameter_ranges = {
+        "categorical_impute_strategy": ["backwards_fill", "forwards_fill"],
+        "numeric_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+        "target_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+    }
+    """{
+        "categorical_impute_strategy": ["backwards_fill", "forwards_fill"],
+        "numeric_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+        "target_impute_strategy": ["backwards_fill", "forwards_fill", "interpolate"],
+    }"""
+    _valid_categorical_impute_strategies = set(["backwards_fill", "forwards_fill"])
+    _valid_numeric_impute_strategies = set(
+        ["backwards_fill", "forwards_fill", "interpolate"]
+    )
+    _valid_target_impute_strategies = set(
+        ["backwards_fill", "forwards_fill", "interpolate"]
+    )
+
+    def __init__(
+        self,
+        categorical_impute_strategy="forwards_fill",
+        numeric_impute_strategy="interpolate",
+        target_impute_strategy="forwards_fill",
```

Comment:
Out of curiosity, why wasn't `interpolate` chosen for the default strategy for the target if the target is a regression type?

Reply:
Great question! That hadn't occurred to me. Do we have a precedent for doing that with other components?

Reply:
Totally, makes sense. We could also consider this when we add the `TimeSeriesImputer` to `make_pipeline`, adding `interpolation` in as the target impute strategy parameter if not already explicitly set.